homeassistant/components/roborock/image.py

```diff
@@ -0,0 +1,147 @@
+        # we will evaluate on the new coordinator data if we should update the cache.
+        if (
+            dt_util.utcnow() - self.image_last_updated
+        ).total_seconds() > IMAGE_CACHE_INTERVAL and self.is_map_valid():
+            self._attr_image_last_updated = dt_util.utcnow()
+        super()._handle_coordinator_update()
+
+    async def async_image(self) -> bytes | None:
+        """Update the image if it is not cached."""
+        if self.is_map_valid():
+            map_data: bytes = await self.cloud_api.get_map_v1()
+            self.cached_map = self._create_image(map_data)
+        return self.cached_map
+
+    def _create_image(self, map_bytes: bytes) -> bytes:
+        """Create an image using the map parser."""
+        parsed_map = self.parser.parse(map_bytes)
+        if parsed_map.image is None:
+            raise HomeAssistantError("Something went wrong creating the map.")
+        img_byte_arr = io.BytesIO()
+        parsed_map.image.data.save(img_byte_arr, format="PNG")
+        return img_byte_arr.getvalue()
+
+
+async def create_coordinator_maps(
+    coord: RoborockDataUpdateCoordinator,
+) -> list[RoborockMap]:
+    """Get the starting map information for all maps for this device. The following steps must be done synchronously.
+
+    Only one map can be loaded at a time per device.
+    """
+    entities = []
+    maps = await coord.cloud_api.get_multi_maps_list()
+    if maps is not None and maps.map_info is not None:
+        cur_map = coord.current_map
+        # This won't be None at this point as the coordinator will have run first.
+        assert cur_map is not None
+        # Sort the maps so that we start with the current map and we can skip the load_multi_map call.
+        maps_info = sorted(
+            maps.map_info, key=lambda data: data.mapFlag == cur_map, reverse=True
+        )
+        for roborock_map in maps_info:
+            # Load the map - so we can access it with get_map_v1
+            if roborock_map.mapFlag != cur_map:
+                # Only change the map and sleep if we have multiple maps.
+                await coord.api.send_command(
+                    RoborockCommand.LOAD_MULTI_MAP, [roborock_map.mapFlag]
+                )
+                # We cannot get the map until the roborock servers fully process the map change.
+                await asyncio.sleep(MAP_SLEEP)
+            # Get the map data
+            api_data: bytes = await coord.cloud_api.get_map_v1()
+            entities.append(
+                RoborockMap(
+                    f"{slugify(coord.roborock_device_info.device.duid)}_map_{roborock_map.name}",
+                    coord,
+                    roborock_map.mapFlag,
+                    api_data,
+                    roborock_map.name,
+                )
+            )
+        if len(maps.map_info) != 1:
+            # Set the map back to the map the user previously had selected so that it does not change the end user's app
```

Comment:
Please break long comments and strings around max 88 characters per line.

Reply:
88 characters including tabs? or 88 characters starting at the beginning of the comment?

Reply:
Count including indentation, so that no line is longer than 88 characters which is what black formats all other lines to follow.